services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/TableMetadata.java

```diff
@@ -72,9 +73,18 @@
      * attribute when using the versioned record extension.
      *
      * @return A collection of all key attribute names for the table.
+     *
+     * @deprecated Use {@link #keyAttributes()} instead.
      */
+    @Deprecated
     Collection<String> allKeys();
 
+    Collection<Index> indices();
```

Comment:
I see I missed some javadocs for these new interface methods. Will add those shortly.